app/views/handlers/relationships/relationship_collector.py

```diff
@@ -14,12 +15,16 @@ def is_location_valid(self):
         return super().is_location_valid()
 
     def handle_post(self):
+        relationship_answer = self.form.data.get(self.relationships_answer_id)
+        relationship = Relationship(
+            self._current_location.list_item_id,
+            self._current_location.to_list_item_id,
+            relationship_answer,
+        )
+        self.relationship_store.add_or_update(relationship)
         self.questionnaire_store_updater.update_relationships_answer(
             relationship_store=self.relationship_store,
-            form_data=self.form.data,
             relationships_answer_id=self.relationships_answer_id,
-            list_item_id=self._current_location.list_item_id,
-            to_list_item_id=self._current_location.to_list_item_id,
         )
```

Comment:
I think we should check if the relationship store is dirty first and instead of waiting until the answer store checks the value against the existing value.

Reply:
Good spot. This was originally done in questionnaire store updater and I didn't notice that when I moved it.